roles/acme/tasks/challenge/dns-01/nsupdate.yml

```diff
@@ -0,0 +1,70 @@
+---
+# include/role 3 - validate challenge
+- name: Validate challenge only if it is created or changed # noqa no-handler
+  when: acme_challenge is changed
+  connection: local
+  delegate_to: localhost
+  vars:
+    relevant_domains: "{{ ([acme_domain.zone] + acme_domain.subject_alt_name | default([])) | ansible.builtin.unique }}"
```

Comment:
I think this needs to change. `acme_domain.zone` will not be in the generated csr and will not be considered as a valid domain for the certificate. Also this is a difference to the other challenge providers and I'd like to keep the basic logic the same.

What need do you see for doing this?

Reply:
Allright, I've changed acme_domain.zone into acme_domain.certificate_name. As one could specify a wildcard here, the *. needs to be replaced in the record (see below). This is also the way how it's done in the other challenge handlers (like autodns.yml).